src/deluge/model/scale/preset_scales.h

```diff
@@ -87,11 +86,13 @@ extern std::array<char const*, NUM_PRESET_SCALES> presetScaleNames;
 // arrays!
 #define OFFICIAL_FIRMWARE_RANDOM_SCALE_INDEX 7
 #define OFFICIAL_FIRMWARE_NONE_SCALE_INDEX 8
-#define CUSTOM_SCALE_WITH_MORE_THAN_7_NOTES 255
+#define CANT_CHANGE_SCALE 255
 #define PRESET_SCALE_RANDOM 254
 #define PRESET_SCALE_NONE 255
 // These offsets allow us to introduce new 7, 6 and 5 note scales in between the existing
 // ones keeping the decreasing order and without breaking backwards compatibility for
 // defaults stored in flash memory.
 #define OFFSET_6_NOTE_SCALE 64
 #define OFFSET_5_NOTE_SCALE 128
+
+const char* getScaleName(uint8_t scale);
```

Comment:
As long as we're touching all the places that pass around scale IDs, maybe it's worth `typedef`'ing it in case we want to make it more strongly typed later.

Reply:
No objections to making it less of a magic index, but I think that instead of adding any more preset scales to the firmware new scales should be on the SD card as scala files, which leads scale id / index being an insufficient identifier. Haven't thought this fully through yet.

Reply:
So that ballooned a bit... Made a separate PR here, https://github.com/SynthstromAudible/DelugeFirmware/pull/2381 but I can update this one as well.